polynote-frontend/polynote/ui.js

```diff
@@ -543,24 +543,21 @@ export class NotebookCellsUI extends UIEventTarget {
         this.configUI = new NotebookConfigUI().setEventParent(this);
         this.el = div(['notebook-cells'], [this.configUI.el]);
         this.el.cellsUI = this;  // TODO: this is hacky and bad (using for getting to this instance via the element, from the tab content area of MainUI#currentNotebook)
-        this.cells = {};
+        this.cells = new Map();
```

Comment:
What's the benefit of using a `Map` here instead of just an object?

Reply:
Hmm, the idea was that since it preserves insertion order and I can use it for `getFirstCell`.

However, now that I think about it the insertion order doesn't necessarily correspond to the order on the page so I guess it's actually broken... oof.

Reply:
Hate to say, but maybe we need both a id -> cell map(/object) *and* a cells array corresponding to display order? At least, if you want to avoid various `querySelector[All]` usages that compute that based on DOM elements.